### ISSUE TITLE
vector allocation-tracing error

### ISSUE THREAD

**Issue by kong62** [vectordotdev/vector#17478]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

```
[#] kubectl logs -f -n vector vector-554fb59747-55rvw --previous
thread 'vector-worker' panicked at 'index out of bounds: the len is 128 but the index is 192', src/internal_telemetry/allocations/mod.rs:106:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2023-05-24T09:17:37.822874Z ERROR source{component_kind="source" component_id=internal_metrics component_type=internal_metrics component_name=internal_metrics}: vector::topology: An error occurred that Vector couldn't handle: the task panicked and was aborted.
2023-05-24T09:17:37.822906Z  INFO vector: Vector has stopped.
2023-05-24T09:17:37.823005Z  INFO vector::topology::running: Shutting down... Waiting on running components. remaining_components="transform_ecs_warning, kafka_k8s, transform_info_first, transform_warning_first, kafka_error, kafka_warning, transform_error, transform_ecs_info_first, internal_metrics, host_metrics, transform_ecs_error_first, transform_ecs_error, prometheus_sink, kafka_info, transform_ecs_info, kafka_ecs, kafka_ecs_info, kafka_ecs_error, transform_info, transform_ecs_warning_first, transform_warning, kafka_ecs_warning" time_remaining="59 seconds left"
```

### Configuration

start:
```text
# vector --allocation-tracing
```

config:
```
  sources:
    internal_metrics:
      type: internal_metrics
      scrape_interval_secs: 30

  sinks:
    prometheus_sink:
      address: 0.0.0.0:9090
      inputs:
        - internal_metrics
        - host_metrics
      type: prometheus_exporter
```
```


### Version

0.30.0

### Debug Output

_No response_

### Example Data

_No response_

### Additional Context

_No response_

### References

_No response_

**Comment by jszwedko** [comment]
Thanks for opening this @kong62 . I think it is a duplicate of https://github.com/vectordotdev/vector/issues/17294 so I'll close this one, but please feel free to follow along there!
